alacritty_terminal/src/term/mod.rs

```diff
@@ -687,6 +687,87 @@ impl VisualBell {
     }
 }
 
+/// LineDamage describes the current damage of a line, represented as two
+/// column coordinates describing the edges of the damage. Damage is valid if
+/// left <= right.
+#[derive(Clone)]
+pub struct LineDamage {
+    pub left: Column,
+    pub right: Column,
+    pub line: Line,
+}
+
+impl LineDamage {
+    /// Create a new invalid LineDamage, which has left=columns+1 and right=0.
+    /// This ensures that later update with min/max of any valid column
+    /// automatically creates a valid line damage without needing any
+    /// conditionals.
+    fn new(line: Line, columns: Column) -> LineDamage {
```

Comment:
This very confusing function btw. From a function called `new` you're expecting to get something valid. Can we just pass invalid arguments or make additional invalid function?

Reply:
We ran rename `new` to `invalid`, but this is the only correct state for a new LineDamage to be constructed in.

I'd very much prefer to keep the invalid construct (inverted left/right) contained within a constructor method like this. Having magic at random callsites is much less readable, and it eould be annoyinf to document N times for N call sites instead of once.

However, now that I think about it, I should probably add an `is_valid` method to fully encapsulate it.

Reply:
Hm, lets just name it somehow meaningful then. This `new` is not fine for something called `LineDamage` imho, invalid also doesn't improve the situation though. What do you think about `placeholder` name, since this line is useless and will be something meaningful afterwards?

Reply:
Ah, based on your name suggestion, maybe there is some confusion about the purpose.

An inverted LineDamage is a clean line, a valid LineDamage is a dirty line. Inverting the bounds, which creates a LineDamage that is logically invalid, is used to indicate that the line is clean in a way that makes it very efficient to dirty without any conditionals.

This, an invalid LineDamage is not a placeholder, but the correct state for untouched lines.

Reply:
I do understand the idea behind it, I just don't know how to name it properly, since its absolutely confusing thing. Oh, may be `undamaged` will be a better name then?